### PR TITLE
refact(gpu): factor out field opencl library

### PR DESCRIPTION
**Pull request by keyvank**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ byteorder = "1"
 log = "0.4.8"
 lazy_static = "1.4.0"
 ocl = { version = "0.19.4", package = "fil-ocl", optional = true }
-itertools = { version = "0.8.0", optional = true }
+ff-cl-gen = { git = "https://github.com/filecoin-project/ff-cl-gen", optional = true }
```
Comment (dignifiedquire) [inline Cargo.toml:28]: lets make a release of that before merging, as we can't publish bellperson to crates.io otherwise

File: Cargo.toml [hunk 2]
```diff
@@ -40,7 +40,7 @@ env_logger = "0.7.1"
 
 [features]
 default = ["groth16", "multicore"]
-gpu = ["ocl", "itertools", "fs2"]
+gpu = ["ocl", "ff-cl-gen", "fs2"]
```
Comment (dignifiedquire) [inline Cargo.toml:43]: why do we need `ocl` as dependency here, if this is all factored out now?

File: Cargo.toml (hunk 1 above)
Comment (keyvank) [reply Cargo.toml:28]: hmm, can I release that from my personal account too? or should we do it with Filecoin's?

File: Cargo.toml (hunk 2 above)
Comment (keyvank) [reply Cargo.toml:43]: only the field library is factored out. fft, multiexp and elliptic curve ops are still here.

**Comment by dignifiedquire** [comment]
@keyvank can you rebase this?

**Comment by keyvank** [comment]
@porcuquine @dignifiedquire `ff-cl-gen` is ready to be released

File: Cargo.toml (hunk 1 above)
Comment (dignifiedquire) [reply Cargo.toml:28]: you can release it from your personal account and then give @cryptonemo and myself at least access on crates.io

**Pull request merged**